### PR TITLE
[SCIM-25] added configuration params to use proxy

### PR DESCRIPTION
**Pull request by andrea-patricelli**

### CODE REVIEW

File: src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java [hunk 1]
```diff
@@ -80,6 +81,17 @@ public class SCIMConnectorConfiguration extends AbstractConfiguration implements
 
     private String scimProvider = SCIMProvider.STANDARD.name();
 
+    // proxy configuration section
+    private String proxyServerType = ProxyServerType.HTTP.value();
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java:85]: I think it is safer to either (a) assume proxy server type is always HTTP or (b) define an `enum` in the connector with the same values as CXF's `enum`

File: src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java [hunk 2]
```diff
@@ -295,6 +307,61 @@ public void setScimProvider(final String scimProvider) {
         this.scimProvider = scimProvider;
     }
 
+    @ConfigurationProperty(displayMessageKey = "proxyServerType.display",
```
Comment (ilgrosso) [inline src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java:310]: Are these messages missing?

**Review by ilgrosso — CHANGES_REQUESTED** [review]

File: src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java (hunk 2 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java:310]: Yes, going to add them

File: src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java (hunk 1 above)
Comment (andrea-patricelli) [reply src/main/java/net/tirasa/connid/bundles/scim/common/SCIMConnectorConfiguration.java:85]: opting for b, since CXF supports also SOCKS proxies

**Review by ilgrosso — APPROVED** [review]

**Pull request merged**